packages/babel-parser/src/parser/util.js

```diff
@@ -147,4 +147,22 @@ export default class UtilParser extends Tokenizer {
       );
     }
   }
+
+  checkYieldAwaitInDefaultParams() {
+    if (
+      this.state.yieldPos &&
+      (!this.state.awaitPos || this.state.yieldPos < this.state.awaitPos)
+    ) {
+      this.raise(
+        this.state.yieldPos,
+        "Yield expression cannot be a default value",
```

Comment:
This error message isn't always accurate:

```js
function* f() {
  (yield) => {}
}
```

```js
function* f() {
  ({ [yield]: x }) => {}
}
```

Reply:
true will change.